bson/src/test/unit/org/bson/codecs/pojo/PojoCustomTest.java

```diff
@@ -262,6 +270,47 @@ public void testCanHandleTopLevelGenericIfHasCodec() {
                 "{'myGenericField': {'$numberLong': '5'}}");
     }
 
+    @Test
+    public void testCustomRegisteredPropertyCodecWithValue() {
```

Comment:
Support for custom containers like `Optional` is now easy with a custom `PropertyCodecProvider`.